app/pods/publications/publication/proofs/index/route.js

```diff
@@ -6,9 +6,25 @@ import { warn } from '@ember/debug';
 
 export class TimelineActivity {
   @tracked activity;
+  @tracked canDeletePieces;
 
   constructor(activity) {
     this.activity = activity;
+    this.initFields();
+  }
+
+  async initFields() {
+    if (this.isProofingActivity) {
+      let pieces = await this.activity.generatedPieces;
+      pieces = pieces.toArray();
+      let publicationActivities = await Promise.all(
+        pieces.map((piece) => piece.publicationActivitiesUsedBy)
+      );
+      publicationActivities = publicationActivities.flatMap((publicationActivities) =>
+        publicationActivities.toArray()
+      );
+      this.canDeletePieces = publicationActivities.length === 0;
```

Comment:
Het is me uit het Jira ticket niet helemaal duidelijk of de delete-flag bepaald wordt per activity of per document. Ik zou per document logischer vinden, maar hier is precies gekozen voor activity. 
Als het op niveau van activity bepaald moet worden, is het waarschijnlijk efficientier (in aantal network requests) om hier een `queryOne` te doen met een filter op de piece ids en op het bestaan van de relatie voor `publicationActivitiesUsedBy`. Als er zo minstens 1 document bestaat, moet delete gedisabled worden voor de hele activiteit.

Reply:
Ik weet ook niet wat hier verwacht wordt. Ik heb het aantal netwerk-requests voorlopig beperkt door deze relatie toe te voegen aan de include. Ik herinner me dat @MikiDi hier een voorkeur voor had in de rapporten-tab.

Ik besef dat per piece deleten soms nuttig kan zijn (vb. wanneer na de proofing-request nog een translation-document wordt geupload). Dit vergt wel wat refactoring. Mocht dit het geval zijn, is mijn voorstel om aan de `<PublicationDocumentsList />` een optioneel yield-block te kunnen meegeven om per piece attributes te kunnen meegeven i.p.v. voor heel de lijst.
vb.
```hbs
<PublicationDocumentsList @rows={{this.pieceRows}} as |pieceRow|>
  <PublicationDocumentItem @onDeletePiece={{if pieceRow.canDelete (fn this.deletePiece pieceRow)}} />
</PublicationDocumentsList>
```
Ik vind dit niet de meest vanzelfsprekende opzet van zo'n component, maar zie niet meteen een andere mogelijkheid. Om dit iets te versimpelen, en het gebruik van row-objecten te vermijden, ben ik nog steeds voorstander van meer gebruik van de `compute` helper in [ember-composable-helpers](https://github.com/DockYard/ember-composable-helpers#compute), maar ik weet dat jullie hier geen voorstander van zijn.

Reply:
Ik denk dat we het hier ingewikkelder maken dan het moet zijn. Johan vat de nood van de gebruikers samen: "Het gaat om de fase vóór het verzenden. Ze hebben gewoon een verkeerd document van de harde schijf naar Kaleidos opgeladen en opgeslagen. Daar moet een vuilbakje komen zodat je dat opladen ongedaan kan maken."

Dus de situatie waarbij een document moet verwijderd worden dat al gebruikt is voor een vertaalaanvraag/drukproefaanvraag gaat zich eigenlijk nooit voordoen.

Wat wel kan is dat er nadat de vertaling/drukproef dan is ontvangen, er een verkeerd document wordt opgeslagen, dat moet dus wel kunnen verwijderd worden.

M.a.w., zolang je altijd documenten kan verwijderen in de panelen met als titel "Vertaling/drukproef ontvangen", is de vraag van de gebruikers voldaan. Is dat nu al zo?